test/tomo/backends/astra_cuda_test.py

```diff
@@ -0,0 +1,197 @@
+    discr_vol_data = discr_vol_space.element(phantom)
+
+    # Angles
+    angle_offset = 0
+    angle_intvl = Interval(0, 2 * np.pi)
+    angle_grid = uniform_sampling(angle_intvl, 8, as_midp=False)
+
+    # Detector
+    dparams = Interval(-6, 6)
+    det_grid = uniform_sampling(dparams, 6)
+
+    # Distances for fanflat geometry
+    src_rad = 100
+    det_rad = 10
+
+    # Create geometry instances
+    geom_p2d = Parallel2dGeometry(angle_intvl, dparams, angle_grid, det_grid)
+    geom_ff = FanFlatGeometry(angle_intvl, dparams, src_rad, det_rad,
+                              angle_grid, det_grid, angle_offset)
+
+    # Projection space
+    proj_rect = angle_intvl.insert(dparams, 1)
+    proj_space = FunctionSpace(proj_rect)
+
+    # `DiscreteLp` projection space
+    npixels = (angle_grid.ntotal, det_grid.ntotal)
+    discr_proj_space = uniform_discr_fromspace(proj_space, npixels,
+                                               dtype='float32')
+
+    # Forward and back projections
+
+    # PARALLEL 2D: forward
+    proj_data_p2d = astra_gpu_forward_projector_call(discr_vol_data, geom_p2d,
+                                                     discr_proj_space)
+    assert proj_data_p2d.shape == npixels
+    assert proj_data_p2d.norm() > 0
+
+    # PARALLEL 2D: backward
+    reco_data_p2d = astra_gpu_backward_projector_call(proj_data_p2d, geom_p2d,
+                                                      discr_vol_space)
+    assert reco_data_p2d.shape == nvoxels
+    assert reco_data_p2d.norm() > 0
+
+    # Fanflat: forward
+    discr_vol_data = discr_vol_space.element(phantom)
+    proj_data_ff = astra_gpu_forward_projector_call(discr_vol_data, geom_ff,
+                                                    discr_proj_space)
+    assert proj_data_ff.shape == npixels
+    assert proj_data_ff.norm() > 0
+
+    # Fanflat: backward
+    reco_data_ff = astra_gpu_backward_projector_call(proj_data_ff, geom_ff,
+                                                     discr_vol_space)
+    assert reco_data_ff.shape == nvoxels
+    assert reco_data_ff.norm() > 0
+
+
+@skip_if_no_astra_cuda
+def test_astra_gpu_projector_call_3d():
+    """Test 3D forward and backward projection functions on the GPU."""
+
+    # `DiscreteLp` volume space
+    vol_shape = (4, 5, 6)
+    discr_vol_space = uniform_discr([-4, -5, -6], [4, 5, 6],
+                                    vol_shape, dtype='float32')
+
+    # Phantom
+    phan = np.zeros(vol_shape)
+    phan[1, 1:3, 1:4] = 1
+
+    # Create an element in the volume space
+    discr_data = discr_vol_space.element(phan)
+
+    # Angles
+    angle_offset = 0
+    angle_intvl = Interval(0, 2 * np.pi)
+    angle_grid = uniform_sampling(angle_intvl, 9, as_midp=False)
+
+    # Detector
+    dparams = Rectangle([-7, -8], [7, 8])
+    det_grid = uniform_sampling(dparams, (7, 8))
+
+    # Parameter for cone beam geometries
+    src_rad = 1000
+    det_rad = 100
+    spiral_pitch_factor = 0.5
+
+    # Create geometries
```

Comment:
One thing per test, make a [pytest fixture](https://pytest.org/latest/fixture.html) if you want to avoid repeating yourself.